aptos-move/aptos-vm-types/src/output.rs

```diff
@@ -108,21 +108,34 @@ impl VMOutput {
         resolver: &impl AggregatorV1Resolver,
     ) -> anyhow::Result<TransactionOutput, VMStatus> {
         let materialized_output = self.try_materialize(resolver)?;
-        debug_assert!(
+        Self::convert_to_transaction_output(materialized_output)
+    }
+
+    /// Same as `try_materialize` but also constructs `TransactionOutput`.
+    pub fn into_transaction_output(self) -> anyhow::Result<TransactionOutput, VMStatus> {
+        let (change_set, fee_statement, status) = self.unpack_with_fee_statement();
+        let materialized_output = VMOutput::new(change_set, fee_statement, status);
+        Self::convert_to_transaction_output(materialized_output)
+    }
+
+    fn convert_to_transaction_output(
+        materialized_output: VMOutput,
+    ) -> anyhow::Result<TransactionOutput, VMStatus> {
+        assert!(
             materialized_output
                 .change_set()
                 .aggregator_v1_delta_set()
                 .is_empty(),
             "Aggregator deltas must be empty after materialization."
         );
-        debug_assert!(
+        assert!(
```

Comment:
Curious, why having an assertion is better? I though we need debug_assert to catch errors in debug builds

Reply:
assert always fails. debug_assert passes in prod